### PR TITLE
Feat ✨ ( #8161) Implement logic to fetch report data part-1

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
### Describe your changes :

I worked on implementing logic to fetch report data on the data insight page.
This PR includes logic to fetch 4 charts data and filter charts data by team, tier and days.

#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :
image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/DataInsightDetail.less [hunk 1]
```diff
@@ -11,8 +11,14 @@
...
+@label-color: #37352f90;
+
 .data-insight-card {
   .ant-card-head {
     border-bottom: none;
   }
 }
+
+.data-insight-label-text {
+  color: @label-color !important;
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/DataInsightDetail.less:23]: `!important` is less preferable. Check if we can avoid it

File: openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/TierInsight.tsx [hunk 2]
```diff
@@ -26,50 +27,95 @@ import {
...
       title={
-        <Typography.Title level={5}>
-          {t('label.data-insight-tier-summary')}
-        </Typography.Title>
+        <>
+          <Typography.Title level={5}>
+            {t('label.data-insight-tier-summary')}
+          </Typography.Title>
+          <Typography.Text className="data-insight-label-text">
+            Display the percentage of datasets with tier by type.
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/TierInsight.tsx:97]: localisation

File: openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/TotalEntityInsight.tsx [hunk 3]
```diff
@@ -26,50 +27,98 @@ import {
...
       title={
-        <Typography.Title level={5}>
-          {t('label.data-insight-total-entity-summary')}
-        </Typography.Title>
+        <>
+          <Typography.Title level={5}>
+            {t('label.data-insight-total-entity-summary')}
+          </Typography.Title>
+          <Typography.Text className="data-insight-label-text">
+            Display the total of datasets by type.
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/TotalEntityInsight.tsx:100]: Localisation

File: openmetadata-ui/src/main/resources/ui/src/constants/DataInsight.constants.ts [hunk 4]
```diff
@@ -34,3 +39,80 @@ export const DATA_INSIGHT_GRAPH_COLORS = [
...
+
+export const DATA_INSIGHT_TAB = {
+  Datasets: 'Datasets',
+  'Web Analytics': 'Web Analytics',
+};
+
+export const DAY_FILTER = [
+  {
+    value: 7,
+    label: 'Last 7 Days',
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/constants/DataInsight.constants.ts:76]: For all the label we need to start localisation strings

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/constants/DataInsight.constants.ts:76]: It's constant so we can't use the locale hook.

File: openmetadata-ui/src/main/resources/ui/src/constants/DataInsight.constants.ts [hunk 5]
```diff
@@ -73,42 +74,42 @@ export const DATA_INSIGHT_TAB = {
 export const DAY_FILTER = [
   {
     value: 7,
-    label: 'Last 7 Days',
+    label: i18n.t('label.day-label', { day: 7 }),
   },
   {
     value: 14,
-    label: 'Last 14 Days',
+    label: i18n.t('label.day-label', { day: 14 }),
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/constants/DataInsight.constants.ts:81]: label.day-label --> label.last-no-of-days

File: openmetadata-ui/src/main/resources/ui/src/constants/DataInsight.constants.ts [hunk 6]
```diff
@@ -73,42 +74,42 @@ export const DATA_INSIGHT_TAB = {
...
   },
   {
     value: 'Tier.Tier3',
-    label: 'Tier3',
+    label: i18n.t('label.tier-label', { tier: 3 }),
   },
   {
     value: 'Tier.Tier4',
-    label: 'Tier4',
+    label: i18n.t('label.tier-label', { tier: 4 }),
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/constants/DataInsight.constants.ts:108]: label.tier-number

File: openmetadata-ui/src/main/resources/ui/src/constants/DataInsight.constants.ts (hunk 5 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/constants/DataInsight.constants.ts:81]: Done.

File: openmetadata-ui/src/main/resources/ui/src/constants/DataInsight.constants.ts (hunk 6 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/constants/DataInsight.constants.ts:108]: Done

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**